packages/hooks/src/hooks/useTokenBalance/useTokenBalance.ts

```diff
@@ -0,0 +1,52 @@
+import React, { useContext } from 'react';
+import { ethers } from 'ethers';
+import { Web3Context } from '../../Provider';
+import ERC20 from './ERC20ABI.json';
+
+/**
+ * @param contractAddress -> the address of the ERC20 token contract
+ * @param ownerAddress -> the address of the owner which balance should be fetched
+
+ */
+export const useTokenBalance = (contractAddress: string, ownerAddress: string) => {
+  const [isLoading, setIsLoading] = React.useState(true);
+  const [balance, setBalance] = React.useState<string>();
+  const [errorMsg, setErrorMsg] = React.useState(false);
```

Comment:
```suggestion
  const [error, setError] = React.useState(false);
```

I think `error` is a better name

Reply:
Is it okay for you @etr2460 ? If so I will rename it to "error"

Reply:
My thought was that this shouldn't be a boolean at all, but instead be an errorMessage and a string type, so that we can pass a better error message back to the caller:
>maybe it's better to make this errorMessage and set the type to string. Then, depending on the error, we can pass a different message up


I can iterate on this in a future PR though